node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/UnixPath.java

```diff
@@ -89,37 +98,29 @@ public void setGroup(String group) {
     }
 
     public Instant getLastModifiedTime() {
-        return uncheck(() -> Files.getLastModifiedTime(path)).toInstant();
+        return getAttributes().lastModifiedTime();
     }
 
-    private PosixFileAttributes getAttributes() {
-        return uncheck(() ->
+    public FileAttributes getAttributes() {
+        PosixFileAttributes attributes = uncheck(() ->
                 Files.getFileAttributeView(path, PosixFileAttributeView.class).readAttributes());
+        return new FileAttributes(attributes);
     }
 
-    @FunctionalInterface
-    private interface SupplierThrowingIOException<T> {
-        T get() throws IOException;
-    }
-
-    private static <T> T uncheck(SupplierThrowingIOException<T> supplier) {
+    public Optional<FileAttributes> getAttributesIfExists() {
         try {
-            return supplier.get();
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-    }
+            return Optional.of(getAttributes());
```

Comment:
Shouldn't this `return ifExists(this::getAttributes);`? Although they do catch different exception.

Reply:
Thanks! Looks like I forgot to switch.